src/dialects/Dialect.ts

```diff
@@ -22,6 +22,20 @@ export interface ITableMetadata {
     comment?: string;
 }
 
+export enum CONSTRAINT_TYPES {
```

Comment:
Are these types common to all the Dialects?
If I assuming correctly these are constraint for foreign keys, so we should probably name it something like `FOREIGN_KEY_CONSTRAINTS`. Also for `enum` we should probably use more verbose naming, using constant case:

```ts
export enum FOREIGN_KEY_CONSTRAINTS {
    CASCADE = 'CASCADE',
    SET_DEFAULT = 'SET DEFAULT',
    SET_NULL = 'SET NULL',
    RESTRICT = 'RESTRICT',
    NO_ACTION = 'NO ACTION',
}
```

Reply:
I don't know about other dialects, this is for postgresql.  The keys are what is coming from postgresql already and then how it should be mapped so you can't change it